src/textual/reactive.py

```diff
@@ -42,6 +42,13 @@
 ReactableType = TypeVar("ReactableType", bound="DOMNode")
 
 
+class _Mutated:
+    """A wrapper to indicate a value was mutated."""
+
+    def __init__(self, value: Any) -> None:
```

Comment:
We should probably update the `always_update` docstring in reactive since we're adding a case where it's now ignored.

Reply:
I don't think that's the case. This wrapper is only ever used in binding to force an update, so it wouldn't change the behavior of `always_update`...

Reply:
Not sure if we are on the same wavelength.

When you call `mutate_reactive`, you are basically saying the value has changed. So I would expect everything to behave as if it has. All the watchers should fire, including any attached via `data_bind`.

Other than that change, everything will behave as it always did.

Reply:
> When you call mutate_reactive, you are basically saying the value has changed. So I would expect everything to behave as if it has. All the watchers should fire, including any attached via data_bind.

I'm talking about the case where you call `mutate_reactive` but the value _hasn't_ changed and `always_update=False`.

When I set `always_update=False` I expect that if the value of the reactive before and after the change is the same, then the watcher will _not_ fire. However, when calling `mutate_reactive` it sounds like that is being ignored and the watcher will fire.

If I have a reactive defined like this:

```python
active_users = reactive(["darren", "will"], always_update=False)
```

And some part of my code "mutates" the list:

```python
active_users[0] = "darren"
self.mutate_reactive(Foo.active_users)
```

The value before and the value after happen to be equal after the mutation. Since `always_update=False`, I would expect the watchers to _not_ fire because the new value equals the old value. This is implied by the `always_update` docstring:

```
always_update: Call watchers even when the new value equals the old value.
```